### ISSUE TITLE
[seq-preserve] causes exceptions when copied inadvertently

### ISSUE THREAD

**Issue by Suchiman** [datalust/seq-releases#586]
Imagine you have a EmailPlus instance installed and you decide you need another Instance.
The fastest way seems to be to go to the old instance, `edit as JSON`, copy the contents, create a new instance, edit as JSON, paste, save.
Now when you do this, seq punishes you for cheating with a red banner 😁

If you remove
```json
"Password": "[seq-preserve]"
```
from the JSON, then saving works as expected.

The log says
```
{"@t":"2017-06-12T14:35:55.1240593Z","@mt":"Error serving {RequestUrl} (token: {ErrorToken})","@l":"Error","@x":"System.NullReferenceException: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.\r\n   bei Seq.Server.Web.Api.AppInstances.AppInstancesModule.MapForStorage(HostedApp app, Dictionary`2 previousSettings, Dictionary`2 postedSettings)\r\n   bei Seq.Server.Web.Api.AppInstances.AppInstancesModule.ToExistingDocument(AppInstanceEntity entity, AppInstance document)\r\n   bei Seq.Server.Web.EntityResourceModule`2.ToNewDocument(TEntity entity)\r\n   bei Seq.Server.Web.Api.AppInstances.AppInstancesModule.ToNewDocument(AppInstanceEntity entity)\r\n   bei Seq.Server.Web.EntityResourceModule`2.Create()\r\n   bei Nancy.Routing.Route.<>c__DisplayClass4.<Wrap>b__3(Object parameters, CancellationToken context)","RequestUrl":"http://app-drive.assfinet.de:5341/api/appinstances/?runOnExisting=false","ErrorToken":"48c39b7d776d4883b0ef02bcc360a1d6","RequestId":"c99ae614-3d46-40fd-a011-ebd1344eecef"}
```

Version used was `4.0.60`

**Comment by nblumhardt** [comment]
Thanks for the report! I'll have a think about how to make this experience nicer.

**Comment by nblumhardt** [comment]
Fix coming in 4.1. Also fixed some quirky behavior whereby pressing "Done" or "Cancel" on the raw editor would save the app instance being edited.
